### PR TITLE
Refactor/adjust tests

### PR DESCRIPTION
**Pull request by ASands17**
What does this PR do?
- Adds complete class structures for Traveler.js, Trip.js, and Trips.js.
- Adds tests to accompany Traveler.js, Trip.js, and Trips.js classes.
- Creates (very) initial DOM layout using HTML and CSS.
- Creates fetch GET request to trips, destinations, and travelers endpoints.

What (if any) features are you implementing?
- Traveler request will return user's first name.
- Trip class has methods to determine a trip destination, calculate its cost, and determine if it is a current, past, or upcoming trip.
- Trips class has methods to create Trip instances, get all trips for a single user, and get cost of all approved trips.

What (if anything) did you refactor?
- Tests were refactored as methods and class structures were changed.

Were there any issues that arose?
- N/A

Is there anything that you need from your teammate?
- If you could pay special attention to my naming conventions and/or testing in this review, I would be appreciative! 

Any other comments, questions, or concerns?
- N/A

### CODE REVIEW

File: src/API-fetch.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+//FETCH REQUESTS GET
+
+export function getTrips() {
```
Comment (tfoxcollis) [inline src/API-fetch.js:3]: Refactor suggestion here, you can create 1 function to handle all 3 of these requests.  Passing in the endpoint ie "trips", "destinations", "travelers" and then interpolating that endpoint into the fetch url.

File: src/Traveler.js [hunk 2]
```diff
@@ -2,14 +2,11 @@ class Traveler {
   constructor(singleTraveler) {
     this.id = singleTraveler.id;
     this.name = singleTraveler.name;
-    this.travelerType = singleTraveler.travelerType;
```
Comment (tfoxcollis) [inline src/Traveler.js:5]: Just curious, why remove the travelerType attribute from constructor?

File: test/Traveler-test.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
+import { expect } from "chai";
+import Traveler from "../src/Traveler";
+
+describe("Traveler", () => {
+  let traveler1, traveler2, traveler3;
+  let traveler1Data, traveler2Data, traveler3Data;
+
+  beforeEach(() => {
+    traveler1Data = {
```
Comment (tfoxcollis) [inline test/Traveler-test.js:9]: Another suggestion to help clean up, you can move the test data into a mock file.  And then call on that data in each of your test files.  This helped me clean up the beforeEach, and also 1 source of data made it easier to correct changes as I updated my src files.

File: src/Trip.js [hunk 4]
```diff
@@ -1,12 +1,71 @@
...
+    this.destinationObj;
+    this.isCurrent;
+    this.isUpcoming;
+    this.isPast;
+    this.destinationData = destinationData;
+
+    this.checkIfCurrentTrip();
+    this.checkIfUpcomingTrip();
+    this.checkIfPast();
+    this.determineDestination();
```
Comment (tfoxcollis) [inline src/Trip.js:18]: The first 3 lines of these checkif functions look similar- maybe they can be broken into a helper function?

File: src/Traveler.js (hunk 2 above)
Comment (ASands17) [reply src/Traveler.js:5]: Good question! I'm not using that data in this iteration, so I decided to take it out. If I end up using it down the line, I'll just add that piece back in.

File: src/scripts.js [hunk 5]
```diff
@@ -1,11 +1,79 @@
...
+    globalDestinations = values[2];
+    startFnOffHere();
+
+    //Start DOM manip from fn above (will rename)
+  })
+}
+
+//FUNCTIONS
+function startFnOffHere() {
+  console.log("globalData in method", globalTravelers, globalTrips, globalDestinations);
```
Comment (JSweet314) [inline src/scripts.js:67]: Don't forget to remove console.logs :) 

Are you making best user of `debugger` statements and browser dev tools?

File: test/Traveler-test.js (hunk 3 above)
Comment (JSweet314) [reply test/Traveler-test.js:9]: If used in multiple test files, definitely worthwhile to extract. Good pattern would be to wrap it in a function and allow property overrides to be passed in so you can create multiple copies with distinct values where needed. (i.e. an instance with `id=1` and another with `id=2`).

File: test/Trip-test.js [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+  });
+
+  it("should have the number of travelers", () => {
+    expect(trip1.travelers).to.equal(1);
+    expect(trip2.travelers).to.equal(5);
+    expect(trip3.travelers).to.equal(3);
+  });
+
+  it("should have the start date of the trip", () => {
+    expect(trip1.date).to.equal("2022/09/16");
```
Comment (JSweet314) [inline test/Trip-test.js:106]: toExist would be sufficient. stronger test even would be to assert the shape of the string (i.e. matches regex pattern for date)

File: test/Trip-test.js [hunk 7]
```diff
@@ -0,0 +1,203 @@
+import { expect } from "chai";
+import Trip from "../src/Trip";
+
+describe("Trip", () => {
```
Comment (JSweet314) [inline test/Trip-test.js:4]: Would suggest reducing number of assertions in this test suite. For tests regarding existence of properties on the Trip class, testing one instance is sufficient. If there is logic that is applied in the constructor, then you would need different instances.

File: src/API-fetch.js [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+  return fetch('http://localhost:3001/api/v1/trips')
+  .then(res => res.json())
+  .catch(error => alert('ERROR'));
+}
+
+export function getDestinations() {
+  return fetch('http://localhost:3001/api/v1/destinations')
+  .then(res => res.json())
+  .catch(error => alert('ERROR'));
```
Comment (JSweet314) [inline src/API-fetch.js:12]: Is error handling on your project roadmap? 😄

**Pull request merged**